src/components/include/hypercall.h

```diff
@@ -0,0 +1,183 @@
+#ifndef HYPERCALL_H
+#define HYPERCALL_H
+
+#include <cos_kernel_api.h>
+#include <cos_defkernel_api.h>
+
+enum hypercall_cntl {
+	HYPERCALL_COMP_INIT_DONE = 0,
+	HYPERCALL_COMP_INFO_GET, /* packed! <retval>, <pgtbl, captbl>, <compcap, parent_spdid> */
+	HYPERCALL_COMP_INFO_NEXT, /* iterator to get comp_info */
+	HYPERCALL_COMP_FRONTIER_GET, /* get current cap frontier & vaddr frontier of spdid comp */
+
+	HYPERCALL_COMP_COMPCAP_GET,
+	HYPERCALL_COMP_CAPTBLCAP_GET,
+	HYPERCALL_COMP_PGTBLCAP_GET,
+	HYPERCALL_COMP_CAPFRONTIER_GET,
+
+	HYPERCALL_COMP_INITTHD_GET,
+	HYPERCALL_COMP_CHILD_NEXT,
+
+	HYPERCALL_NUMCOMPS_GET,
+};
+
+static inline int
+hypercall_comp_child_next(spdid_t c, spdid_t *child, comp_flag_t *flags)
+{
+	word_t r2 = 0, r3 = 0;
+	int ret;
+
+	ret = cos_sinv_rets(BOOT_CAPTBL_SINV_CAP, 0, HYPERCALL_COMP_CHILD_NEXT, c, 0, &r2, &r3);
+	if (ret < 0) return ret;
+	*child = (spdid_t)r2;
+	*flags = (comp_flag_t)r3;
+
+	return ret; /* remaining child spds */
+}
+
+static inline int
+hypercall_comp_init_done(void)
+{
+	/*
+	 * to be used only by the booter child threads
+	 * higher-level components use, schedinit interface to SINV to parent for init
+	 */
+	return cos_sinv(BOOT_CAPTBL_SINV_CAP, 0, HYPERCALL_COMP_INIT_DONE, 0, 0);
+}
+
+/* Note: This API can be called ONLY by components that manage capability resources */
+static inline int
+hypercall_comp_initthd_get(spdid_t spdid, int is_sched, thdcap_t *thdslot, arcvcap_t *rcvslot, tcap_t *tcslot)
```

Comment:
This is only used by capmgr. It really only needs tcap and rcvcap of really `root sched`. so it can create threads with those rcvcap as parents etc. 

I'll rename it.